android/uhabits-core/src/main/java/org/isoron/uhabits/core/ui/screens/habits/show/ShowHabitBehavior.java

```diff
@@ -58,12 +59,29 @@ public void onEditHistory()
 
     public void onToggleCheckmark(Timestamp timestamp, int value)
     {
-        commandRunner.execute(
-            new CreateRepetitionCommand(habitList, habit, timestamp, value), null);
+        if (habit.isNumerical()) {
+            CheckmarkList checkmarks = habit.getCheckmarks();
+            double oldValue = checkmarks.getValues(timestamp, timestamp)[0];
+
+            screen.showNumberPicker(oldValue / 1000, habit.getUnit(), newValue ->
+            {
+                newValue = Math.round(newValue * 1000);
+                commandRunner.execute(
+                        new CreateRepetitionCommand(habitList, habit, timestamp, (int) newValue),
+                        habit.getId());
+            });
```

Comment:
This is a repetition of the code in `ListHabitBehavior`. Please create a common method.

Reply:
Hi @iSoron, could you share a suggestion for this change?